autoload/go/template.vim

```diff
@@ -10,12 +11,19 @@ function! go#template#create() abort
   let l:package_name = go#tool#PackageName()
 
   " if we can't figure out any package name(no Go files or non Go package
-  " files) from the directory create the template
-  if l:package_name == -1
+  " files) from the directory create the template or use the cwd
+  " as the name
+  if l:package_name == -1 && l:go_template_no_file != 1
     let l:template_file = get(g:, 'go_template_file', "hello_world.go")
     let l:template_path = go#util#Join(l:root_dir, "templates", l:template_file)
     exe '0r ' . fnameescape(l:template_path)
     $delete _
+  elseif l:package_name == -1 && l:go_template_no_file == 1
+    " cwd is now the dir of the package
+    let l:path = fnamemodify(getcwd(), ':t')
```

Comment:
> Using package instead of directory in the docs means that the docs would have to say something like, "use the package name for the name for new packages".

I disagree, it's up to us how to write it down. I went over again and here are the fixed doc sentences:


```
When a new Go file is created, vim-go automatically fills the buffer content
with a Go code template. By default the template under
`templates/hello_world.go` is used. This can be changed with the
|'g:go_template_file'| setting.

If the new file is created in an already prepopulated package (with other Go
files), in this case a Go code template with only the Go package declaration
(which is automatically determined according to the current package) is added.

To use always the package name instead of the template (`tempaltes/hello_world.go`)
when new Go file is created, enable the |'g:go_template_use_pkg'| setting.

By default it is enabled.
>
  let g:go_template_autocreate = 1
<
```

and 

```
Specifies that, rather than using a template, the package name is used 
if a new Go file is created. Checkout |'g:go_template_autocreate'| for more 
info. By default the `hello_world.go` file is used.

>
  let g:go_template_use_pkg = 0
<
```

As you see, it's not complicated and it makes sense to use it this way. It's understandable and there is nowhere the notion of directory. If you just update the docs with the ones above and change the variable name to `go_template_use_pkg` we're good to go.